File: Intent/app/src/main/java/com/shsnk/intent/MainActivity.java
package com.shsnk.intent;

import androidx.appcompat.app.AppCompatActivity;

import android.annotation.SuppressLint;
import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity implements View.OnClickListener {

    @Override
    public void onClick(View view) {
        Toast.makeText(this, "Welcome", Toast.LENGTH_SHORT).show();
        Intent intent = new Intent(this,MainActivity2.class);
        txt=findViewById(R.id.editText);
        String str = txt.getText().toString();
        intent.putExtra(EXTRA_MESSAGE,str);
        startActivity(intent);
    }
    Button btn;
    EditText txt;
    public static final String EXTRA_MESSAGE = "com.shsnk.intent.MESSAGE";
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        btn=findViewById(R.id.button);
        btn.setOnClickListener(this);
    }
}

File: KotlinDice/app/src/main/java/com/shsnk/kotlindice/MainActivity.kt
package com.shsnk.kotlindice

import android.os.Bundle
import android.widget.Button
import android.widget.TextView
import android.widget.Toast
import android.widget.ImageView
import androidx.appcompat.app.AppCompatActivity

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val btn : Button=findViewById(R.id.buttonRoll)
        btn.setOnClickListener {
        //temporary text
            var toast=Toast.makeText(this,"Dice Rolled", Toast.LENGTH_SHORT)
            toast.show()
        //permanent text show
            var side1=dice(6)
            var RandomNumber = side1.roll()
            var text2 : TextView= findViewById(R.id.textView2)
            text2.text=RandomNumber.toString()
            var diceImage : ImageView =findViewById(R.id.dice_1)
                when(RandomNumber)
                {
                    1-> diceImage.setImageResource(R.drawable.dice_1)
                    2-> diceImage.setImageResource(R.drawable.dice_2)
                    3-> diceImage.setImageResource(R.drawable.dice_3)
                    4-> diceImage.setImageResource(R.drawable.dice_4)
                    5-> diceImage.setImageResource(R.drawable.dice_5)
                    6-> diceImage.setImageResource(R.drawable.dice_6)
                }

        // textShow.text=RandomNumber.toString() //Important to convert number into text
            /*Dice No. 2
            var side2=dice(12)
            var RandomNumber2=side2.roll()
            var dice2 : TextView = findViewById(R.id.textView2)
                dice2.text=RandomNumber2.toString()
             */
        }

    }
}
class dice(var Side:Int)
{
    fun roll():Int
    {
        return ((1..Side).random())
    }
}